src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -279,50 +267,25 @@ private void updateDeviceGroups(WhatHappened whatHappened, Topics deviceGroupsTo
         }
     }
 
-    private void updateCAType(Topic topic) {
+    private void updateCAType() {
+        String thingName = Coerce.toString(deviceConfiguration.getThingName());
         try {
-            List<String> caTypeList = Coerce.toStringList(topic);
-            logger.atDebug().kv("CA type", caTypeList).log("CA type list updated");
-
-            if (Utils.isEmpty(caTypeList)) {
-                logger.atDebug().log("CA type list null or empty. Defaulting to RSA");
-                certificateManager.update(getPassphrase(), CertificateStore.CAType.RSA_2048);
-            } else {
-                if (caTypeList.size() > 1) {
-                    logger.atWarn().log("Only one CA type is supported. Ignoring subsequent CAs in the list.");
-                }
-                String caType = caTypeList.get(0);
-                certificateManager.update(getPassphrase(), CertificateStore.CAType.valueOf(caType));
-            }
+            certificateManager.update(caConfiguration.getCaPassphrase(), caConfiguration.getCaType());
+            certificateManager.uploadCoreDeviceCAs(thingName);
```

Comment:
the upload will need to be taken out of this method eventually. We can't wait in this method for the upload to complete (as it may never complete).